### ISSUE TITLE
ncsixel_as_rgba: incorrect state is set following '-'

### ISSUE THREAD

**Issue by waveplate** [dankamongmen/notcurses#2787]
oi, there seems to be another bug in `ncsixel_as_rgba` !

```
      }else if(*sx == '$'){
        x = 0;
        state = STATE_WANT_DATA;
      }else if(*sx == '-'){
        x = 0;
        y += 6;
        state = STATE_WANT_HASH;
      }else{
```

setting the state to `STATE_WANT_DATA` after the `-` command fixed it for me

or more concisely, since the state is already `STATE_WANT_DATA` at this point

```
      }else if(*sx == '$'){
        x = 0;
      }else if(*sx == '-'){
        x = 0;
        y += 6;
      }else{
```

the following sixel images will trigger the bug:

[laughing.txt](https://github.com/user-attachments/files/15930780/laughing.txt)
[red40.txt](https://github.com/user-attachments/files/15930781/red40.txt)

been using `notcurses` for a little while now -- really great project :)

**Comment by dankamongmen** [comment]
good find! as you can see, this code has not been heavily used nor properly tested =].

i'm glad to hear you're liking the project. i'm hoping to do a release very soon that will wrap up these fixes.

**Comment by dankamongmen** [comment]
```c
process_escape:2219:walk result on 99 (c): 1 386
handoff_initial_responses_late:1198:handing off initial responses
block_on_input:2457:blocking on input availability
block_on_input:2514:waiting on 1 fds (ibuf: 0/8192)
ncplane_new_internal:692:created new 64x132 plane "std" @ 0x0
ncsixel_as_rgba:172:invalid rle 1 + 1 > 1
ncvisual_from_sixel:809:failed converting sixel to rgba
===============================================================================
/home/dank/src/dankamongmen/notcurses/src/tests/sixels.cpp:17:
TEST CASE:  Sixels
  LoadBiggerSixel

/home/dank/src/dankamongmen/notcurses/src/tests/sixels.cpp:46: FATAL ERROR: REQUIRE( ncv ) is NOT correct!
  values: REQUIRE( nullptr )

===============================================================================
[doctest] test cases:  1 |  0 passed | 1 failed | 45 skipped
[doctest] assertions: 18 | 17 passed | 1 failed |
[doctest] Status: FAILURE!
[schwarzgerat](1) $ 
```
i'm reproducing this in unit tests now.

**Comment by dankamongmen** [comment]
Closed by #2788. Thanks a bunch, @waveplate !
